### PR TITLE
Make mz_kafka_sinks a local input

### PR DESCRIPTION
**Pull request by justinj**
A little unsure about the organization here, lots of unfortunate overlap with the logging views, probably what would be best would be to unify those as a single enum that dispatches either to a logging view or a local input, but open to discussion!

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3535)

### CODE REVIEW

**Comment by benesch** [comment]
This all seems good to me—not fussed about the duplication! The only thing that's a little bit scary is the fact that we might accidentally reuse a system ID, but that's easy enough to solve after this lands.

These errors look legit though?

```
caused by: ERROR: Timestamp (6) is not valid for all inputs: [(System(32), Some(Antichain { elements: [15000] }))]
````

**Comment by justinj** [comment]
@frankmcsherry I implemented the fix we discussed, it seems to work, from what I can tell. What do you think of this?

File: src/coord/src/coord.rs [hunk 1]
```diff
@@ -312,6 +332,7 @@ where
     }
 
     pub fn get_ts(&mut self) -> Timestamp {
+        self.need_advance = true;
```
Comment (frankmcsherry) [inline src/coord/src/coord.rs:335]: I'd love to have comments here, esp on a public method. What does this do, who should call it, what should they expect, etc.

File: src/coord/src/coord.rs [hunk 2]
```diff
@@ -540,18 +561,25 @@ where
...
-                    &mut self.broadcast_tx,
-                    SequencedCommand::AdvanceAllLocalInputs {
-                        advance_to: next_ts,
-                    },
-                );
-                self.closed_up_to = next_ts;
+            // TODO(justin): this is pretty hacky, and works more-or-less because this frequency
+            // lines up with that used in the logging views.
+            const FREQUENCY: u64 = 1_000;
+            if needed || next_ts / FREQUENCY > self.closed_up_to / FREQUENCY {
```
Comment (frankmcsherry) [inline src/coord/src/coord.rs:573]: So the only thing that worries me here is that `--logging-frequency` is a thing we can change at the command line. By default it is 1s, but if someone dials it up or down I don't know what will happen. We can probably punt that off to the future, but it's the sort of thing that will accrete with time. I don't have anything better (maybe steal the same argument from `--logging-frequency` but wanted to surface shared concern.

File: src/coord/src/coord.rs [hunk 3]
```diff
@@ -1854,6 +1882,21 @@ where
         self.create_sink_dataflow(name.to_string(), id, sink.from, connector)
     }
 
+    fn update_catalog_view(&mut self, v: CatalogView, row: Row, diff: isize) {
```
Comment (frankmcsherry) [inline src/coord/src/coord.rs:1885]: Would be great to document. I suspect that we will want to change this soon to support multiple updates. I suspect it is ergonomic at the moment owing to sinks not being high churn, but for anything else we'll want to bundle updates together in `SequencedCommand::Insert` messages.

File: src/coord/src/coord.rs (hunk 1 above)
Comment (justinj) [reply src/coord/src/coord.rs]: Good call—added a comment and made this non-public.

File: src/coord/src/coord.rs (hunk 2 above)
Comment (justinj) [reply src/coord/src/coord.rs]: Plumbed that value through to here as well. Kinda clunky but I think better!

File: src/coord/src/coord.rs (hunk 3 above)
Comment (justinj) [reply src/coord/src/coord.rs:1897]: Added a comment. We can extend it when we use it for that, I think?

**Comment by justinj** [comment]
I think so—I added some comments, I think most things will be caught by the type system.

**Pull request merged**